keps/sig-node/3721-support-for-env-files/README.md

```diff
@@ -0,0 +1,559 @@
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+Automated tests will cover the scenarios with and without the changes proposed
+on this KEP. As defined under [Version Skew Strategy](#version-skew-strategy),
+we are assuming the cluster may have kubelets with older versions (without
+this KEP' changes), therefore this will be covered of following new tests:
+
+1. When the kubelet is upgraded, the env files will be instantiated in the
+   container. On downgrade, the env files will be ignored but the pod will still
+   run. On upgrade, the env files should be instantiated in the container again.
+
+2. When the apiserver is upgraded, new envs will be written. But on downgrade,
+   it cannot be written but the existing envs will continue to exist. On
+   upgrade, the envs can again be written.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+The feature is built into the kubelet and api server components. No metric is
+planned at this moment. The way to determine usage is by checking whether the
+pods/containers have a `FileEnvSource` or `FileKeySelector` fields set.
+
+###### How can someone using this feature know that it is working for their instance?
+
+The user can exec into the running container and check that the environment
+variable is set:
+
+```
+$ kubectl exec -n $NAMESPACE $POD_NAME -- env
+CONFIG_VAR=HELLO
+```
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+N/A
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+N/A
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+N/A
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+No
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+No
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+No
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+No impact to the running workloads
```

Comment:
but feature wouldnt work for new workloads ?

Reply:
If kube-apiserver or etcd is unavailable, then the new workloads won't be created as the request would fail/timeout.

However, the existing workloads should be safe.